lib/src/bluetooth_service.dart

```diff
@@ -122,4 +122,28 @@ class BluetoothService {
       rethrow;
     }
   }
+
+  ///
+  /// Get a list of characteristics from this service.
+  ///
+  /// [uuid] optional uuid.
+  ///
+  /// TODO: Needs Error checking
+  ///
+  ///
+  Future<List<BluetoothCharacteristic>> getCharacteristics(
+      {String? uuid}) async {
+    try {
+      final List<WebBluetoothRemoteGATTCharacteristic> characteristic =
+          await _bluetoothService.getCharacteristics(uuid);
+      List<BluetoothCharacteristic> characteristicList = [];
+      characteristic.forEach((element) {
+        characteristicList.add(BluetoothCharacteristic(element));
+      });
+      return characteristicList;
+    } catch (e) {
+      final error = e.toString().trim();
+      rethrow;
+    }
```

Comment:
The `getCharacteristics` method gives a list of errors that may be thrown and then need to be converted from a JS exception (just a string) to an actual dart exception so it's easier to handle them.

See the comment for the possible exceptions

https://github.com/jeroen1602/flutter_web_bluetooth/blob/efc36ab59fda9026f0c75e107919e73173e92e99/lib/web/web_bluetooth_remote_gatt_service.dart#L153-L183

And `getCharacteristic` for an example how they are handled (the code can probably just be copy pasted over)

https://github.com/jeroen1602/flutter_web_bluetooth/blob/efc36ab59fda9026f0c75e107919e73173e92e99/lib/src/bluetooth_service.dart#L111-L123

Reply:
Hi, the error stuff I was unsure about because uuid is an optional parameter and some of your error handling requires that value. ie ` throw NetworkError.withUUid(uuid); ` & `throw NotFoundError.forService(uuid, this.uuid); `

Reply:
Your current implementation doesn't throw the actual exception but the runtime type. So instead of an instance of `NotFoundError` it will throw an instance of `Type` for the `NotFoundError` class.

As for which constructors to use I have looked at it and this is what I thought of:

For `SecurityError` just use `SecurityError(this.uuid, error);`. So with the UUID of the service where it is thrown and the original message for the possible link url.

For `NetworkError` use `NetworkError.withUUid(this.uuid)` with the UUID of the of the service.

For `NotFoundError` the existing `forService` constructor will need to be changed to make the first parameter nullable (add a ?). The implementation will be almost the same as `forDescriptor`. After that use the constructor as you described above (`NotFoundError.forService(uuid, this.uuid);`.